NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -91,132 +189,135 @@ def run(
             Lower and upper bound of the frequency range in which noise shall be
             added
         """
+
+        # check that or all channels channel.get_frequencies() is identical
+        last_freqs = None
+        for channel in station.iter_channels():
+            if (not last_freqs is None) and (
+                    not np.allclose(last_freqs, channel.get_frequencies(), rtol=0, atol=0.1 * units.MHz)):
+                logger.error("The frequencies of each channel must be the same, but they are not!")
+                return
+
+            last_freqs = channel.get_frequencies()
+
+        freqs = last_freqs
+        d_f = freqs[2] - freqs[1]
+
         if passband is None:
-            passband = [10 * units.MHz, 1000 * units.MHz]
-        self.__gdsm = pygdsm.pygsm.GlobalSkyModel()
+            passband = [10 * units.MHz, 1100 * units.MHz]
+        passband_filter = (freqs > passband[0]) & (freqs < passband[1])
+
         site_latitude, site_longitude = detector.get_site_coordinates(station.get_id())
-        site_location = astropy.coordinates.EarthLocation(lat=site_latitude * astropy.units.deg, lon=site_longitude * astropy.units.deg)
+        site_location = astropy.coordinates.EarthLocation(lat=site_latitude * astropy.units.deg,
+                                                          lon=site_longitude * astropy.units.deg)
         station_time = station.get_station_time()
-        station_time.format = 'iso'
-        noise_temperatures = np.zeros((len(self.__interpolaiton_frequencies), healpy.pixelfunc.nside2npix(self.__n_side)))
+
         local_cs = astropy.coordinates.AltAz(location=site_location, obstime=station_time)
         solid_angle = healpy.pixelfunc.nside2pixarea(self.__n_side, degrees=False)
-        pixel_longitudes, pixel_latitudes = healpy.pixelfunc.pix2ang(self.__n_side, range(healpy.pixelfunc.nside2npix(self.__n_side)), lonlat=True)
+
+        pixel_longitudes, pixel_latitudes = healpy.pixelfunc.pix2ang(self.__n_side,
+                                                                     range(healpy.pixelfunc.nside2npix(self.__n_side)),
+                                                                     lonlat=True)
         pixel_longitudes *= units.deg
         pixel_latitudes *= units.deg
-        galactic_coordinates = astropy.coordinates.Galactic(l=pixel_longitudes * astropy.units.rad, b=pixel_latitudes * astropy.units.rad)
+
+        galactic_coordinates = astropy.coordinates.Galactic(l=pixel_longitudes * astropy.units.rad,
+                                                            b=pixel_latitudes * astropy.units.rad)
         local_coordinates = galactic_coordinates.transform_to(local_cs)
+
         n_ice = ice.get_refractive_index(-0.01, detector.get_site(station.get_id()))
```

Comment:
And the time! It might be more a problem for my current "processing style" where every station is a dual polarised antenna but in principle if you have several station you want to process at the same time you might be able to save something here!

Reply:
So I tried it with this function:
```
#@functools.lru_cache(maxsize=128)
def get_local_coordinates(coordinates, time, n_side):
    print(coordinates, time, n_side)
    site_latitude, site_longitude = coordinates
    site_location = astropy.coordinates.EarthLocation(lat=site_latitude * astropy.units.deg,
                                                        lon=site_longitude * astropy.units.deg)

    local_cs = astropy.coordinates.AltAz(location=site_location, obstime=time)

    pixel_longitudes, pixel_latitudes = healpy.pixelfunc.pix2ang(n_side,
                                                                    range(healpy.pixelfunc.nside2npix(n_side)),
                                                                    lonlat=True)
    pixel_longitudes *= units.deg
    pixel_latitudes *= units.deg

    galactic_coordinates = astropy.coordinates.Galactic(l=pixel_longitudes * astropy.units.rad,
                                                        b=pixel_latitudes * astropy.units.rad)
    local_coordinates = galactic_coordinates.transform_to(local_cs)

    return local_coordinates
```
Turns out it does not really safe any time because apparently this code is quite fast. However, you might want to add it anyway because it makes the code a little bit more readable?